eval/src/vespa/eval/eval/basic_nodes.h

```diff
@@ -120,9 +121,9 @@ class String : public Leaf {
 public:
     String(const vespalib::string &value_in) : _value(value_in) {}
     bool is_const_double() const override { return true; }
-    double get_const_double_value() const override { return hash(); }
+    double get_const_double_value() const override { return hash2d(_value); }
     const vespalib::string &value() const { return _value; }
-    uint32_t hash() const { return hash_code(_value.data(), _value.size()); }
+    uint64_t hash() const { return hashValue(_value.data(), _value.size()); }
```

Comment:
this needs to agree with get_const_double_value or go away

Reply:
Using get_const_double_value to generate the key and removed hash method.